webhook_server/libs/github_api.py

```
@@ -291,9 +421,255 @@ async def get_pull_request(self, number: int | None = None) -> PullRequest | Non
+        if isinstance(pull_request, PullRequest):
+            owner = pull_request.base.repo.owner.login
+            repo_name = pull_request.base.repo.name
+            # Try GraphQL first for performance
+            try:
+                pr_data = await self.unified_api.get_pull_request(
+                    owner, repo_name, pull_request.number, include_commits=True
+                )
+                # Extract commits from GraphQL response
+                commits_nodes = pr_data.get("commits", {}).get("nodes", [])
+                if not commits_nodes:
+                    raise ValueError(f"No commits found in PR {pull_request.number}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)